include/lbann/layers/learning/base_convolution.hpp

```diff
@@ -730,63 +703,60 @@ class base_convolution_layer : public Layer {
-          CHECK_CUDNN(cudnnGetConvolutionBackwardFilterAlgorithm(cudnn::get_handle(),
-                                                                 input_desc,
-                                                                 gradient_wrt_output_desc,
-                                                                 m_convolution_cudnn_desc,
-                                                                 m_kernel_cudnn_desc,
-                                                                 CUDNN_CONVOLUTION_BWD_FILTER_SPECIFY_WORKSPACE_LIMIT,
-                                                                 workspace_size,
-                                                                 &kernel_gradient_cudnn_algorithm));
+          CHECK_CUDNN(cudnnGetConvolutionBackwardFilterAlgorithm(
+                        cudnn::get_handle(),
+                        input_desc,
+                        gradient_wrt_output_desc,
+                        m_convolution_cudnn_desc,
+                        m_kernel_cudnn_desc,
+                        CUDNN_CONVOLUTION_BWD_FILTER_SPECIFY_WORKSPACE_LIMIT,
+                        workspace_size,
+                        &kernel_gradient_cudnn_algorithm));
           #endif
-          CHECK_CUDNN(cudnnConvolutionBackwardFilter(cudnn::get_handle(),
-                                                     &one,
-                                                     input_desc,
-                                                     local_input.LockedBuffer(),
-                                                     gradient_wrt_output_desc,
-                                                     local_gradient_wrt_output.LockedBuffer(),
-                                                     m_convolution_cudnn_desc,
-                                                     kernel_gradient_cudnn_algorithm,
-                                                     workspace.Buffer(),
-                                                     workspace_size,
-                                                     &zero,
-                                                     m_kernel_cudnn_desc,
-                                                     m_kernel_gradient.Buffer()));
-
+          CHECK_CUDNN(cudnnConvolutionBackwardFilter(
+                        cudnn::get_handle(),
+                        &gradient_scale,
+                        input_desc,
+                        local_input.LockedBuffer(),
+                        gradient_wrt_output_desc,
+                        local_gradient_wrt_output.LockedBuffer(),
+                        m_convolution_cudnn_desc,
+                        kernel_gradient_cudnn_algorithm,
+                        workspace.Buffer(),
+                        workspace_size,
+                        &dst_scale,
+                        m_kernel_cudnn_desc,
+                        kernel_gradient.Buffer()));
         }
-
+      } else {
+        El::Scale(dst_scale, kernel_gradient);
```

Comment:
Should this be dst_scale or gradient_scale here.  Earlier in this code block the gradient_scale is divided by the effective_mini_batch_size, and dst_scale is not.

Reply:
`dst_scale` is correct. In this route, the layer's computed gradient is 0, so it would be meaningless to use `gradient_scale`. We still need to apply the correct scaling factor to the returned gradient buffer, because the rank will still participate in the allreduce. Compare with the `add_to_gradient` method.

(This confusion is why I originally had this hidden within the optimizer class.)